### PR TITLE
Read build from manifest

### PR DESCRIPTION
**Pull request by teresaromero**
Fixes #

## Proposed changes

- Add Build to new manifest declaration
- Parse to BuildParams struct the build options set at manifest for each image

### CODE REVIEW

File: pkg/model/manifest.go [hunk 1]
```diff
@@ -13,21 +13,24 @@
...
-	Destroy []string     `json:"destroy,omitempty" yaml:"destroy,omitempty"`
-
-	Type     string `yaml:"-"`
-	Filename string `yaml:"-"`
+	Name     string        `json:"name,omitempty" yaml:"name,omitempty"`
+	Icon     string        `json:"icon,omitempty" yaml:"icon,omitempty"`
+	Deploy   *DeployInfo   `json:"deploy,omitempty" yaml:"deploy,omitempty"`
+	Dev      ManifestDevs  `json:"devs,omitempty" yaml:"devs,omitempty"`
+	Destroy  []string      `json:"destroy,omitempty" yaml:"destroy,omitempty"`
+	Build    ManifestBuild `json:"build,omitempty" yaml:"build,omitempty"`
```
Comment (pchico83) [inline pkg/model/manifest.go:27]: is it possible to reuse `BuildInfo`?
https://github.com/okteto/okteto/blob/fd03d2c861691e15320dd99f63c97af18674cda7/pkg/model/dev.go#L116

File: pkg/model/build.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+	Image     string   `json:"image,omitempty" yaml:"image,omitempty"`
+	Context   string   `json:"context,omitempty" yaml:"context,omitempty"`
+	File      string   `json:"dockerfile,omitempty" yaml:"dockerfile,omitempty"`
+	Target    string   `json:"target,omitempty" yaml:"target,omitempty"`
+	Args      []string `json:"args,omitempty" yaml:"args,omitempty"`
+	Secrets   []string `json:"secrets,omitempty" yaml:"secrets,omitempty"`
+	CacheFrom []string `json:"cacheFrom,omitempty" yaml:"cacheFrom,omitempty"`
+}
+
+type BuildParams struct {
```
Comment (pchico83) [inline pkg/model/build.go:13]: is it possible to reuse `BuildInfo`?
https://github.com/okteto/okteto/blob/fd03d2c861691e15320dd99f63c97af18674cda7/pkg/model/dev.go#L116

File: pkg/model/manifest_test.go [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+		name              string
+		manifestYAML      []byte
+		manifestNotExists bool
+		expectedErr       bool
+		expectedManifest  *ManifestBuild
+	}{
+		{
+			name: "exists-manifest",
+			manifestYAML: []byte(`build:
+  firstImage: ./first
```
Comment (pchico83) [inline pkg/model/manifest_test.go:21]: I would use a sample with a value for every field supported in the build section

File: pkg/model/serializer_test.go [hunk 4]
```diff
@@ -1778,3 +1778,41 @@ compose:
...
+	tests := []struct {
+		name            string
+		buildManifest   []byte
+		expected        ManifestBuild
+		isErrorExpected bool
+	}{
+		{
+			name: "build params at manifest",
+			buildManifest: []byte(`firstBuild: ./first
+secondBuild:
```
Comment (pchico83) [inline pkg/model/serializer_test.go:1792]: I would use a sample with a value for every field supported in the build section

File: pkg/model/serializer.go [hunk 5]
```diff
@@ -716,7 +716,7 @@ func (d *ManifestDevs) UnmarshalYAML(unmarshal func(interface{}) error) error {
 }
 
 func isManifestFieldNotFound(err error) bool {
-	manifestFields := []string{"devs", "name", "icon", "variables", "deploy", "destroy"}
+	manifestFields := []string{"devs", "name", "icon", "variables", "deploy", "destroy", "build"}
```
Comment (pchico83) [inline pkg/model/serializer.go:719]: @jLopezbarb based on the spec, it should be `dev`, not `devs`:
https://www.notion.so/okteto/Okteto-DEV-915b72bf1a8f4275879974e544a2d614#5ee9c33a37484727a72f5607912f6bfb

**Comment by teresaromero** [comment]
re-opened with new branch name

File: pkg/model/build.go (hunk 2 above)
Comment (teresaromero) [reply pkg/model/build.go:13]: the keys for this struct are different, for example, image is called Name, should i add Image to BuildInfo ?

File: pkg/model/manifest.go (hunk 1 above)
Comment (teresaromero) [reply pkg/model/manifest.go:27]: Build will still need to be a map for the different images defined, you mean changing the BuildParams struct for each image to BuildInfo?

File: pkg/model/build.go (hunk 2 above)
Comment (pchico83) [reply pkg/model/build.go:13]: Yeap, let's add `Image`. `Name` and where it's used will be deprecated after the build section is released

File: pkg/model/manifest.go (hunk 1 above)
Comment (pchico83) [reply pkg/model/manifest.go:27]: Right

**Pull request closed without merging**